packages/astro/src/content/utils.ts

```diff
@@ -43,21 +42,24 @@ export const msg = {
 		`${collection} does not have a config. We suggest adding one for type safety!`,
 };
 
-export function extractFrontmatterAssets(data: Record<string, any>): string[] {
-	function findAssets(potentialAssets: Record<string, any>): ImageMetadata[] {
-		return Object.values(potentialAssets).reduce((acc, curr) => {
-			if (typeof curr === 'object') {
-				if (curr.__astro === true) {
-					acc.push(curr);
+export function patchFrontmatterAssets(data: Record<string, any>): Record<string, any> {
+	function patchAsset(potentialAssets: Record<string, any>): Record<string, any> {
+		Object.keys(potentialAssets).forEach((entry) => {
+			if (typeof potentialAssets[entry] === 'object' && potentialAssets[entry] !== null) {
+				if (potentialAssets[entry].__astro_asset) {
+					potentialAssets[entry] = `$$ASSET_(await import('${normalizePath(
+						potentialAssets[entry].src
+					)}')).default_ASSET$$`;
```

Comment:
This is super ugly, but I couldn't figure out how to do in a better way. Passing and trying to use `this.load()` here failed in various ways